src/modules/layout/components/HeaderDropDown.js

```diff
@@ -1,23 +1,32 @@
-import React from 'react'
-import PropTypes from 'prop-types'
+// @flow
+
+import * as React from 'react'
 import cx from 'classnames'
 import style from './HeaderDropDown.css'
 import onClickOutside from 'react-onclickoutside'
 
+type PropsType = {
+  children: React.Element<any>,
+  iconSrc: string
+}
+
+type StateType = {
+  dropDownActive: boolean
+}
+
 /**
  * Designed to work as an item of a HeaderActionBar. Once clicked, the child node becomes visible right underneath the
  * Header. Once the user clicks outside, the node is hidden again. Additionally, the inner node gets a
  * closeDropDownCallback through its props to close the dropDown and hide itself.
  */
-class HeaderDropDown extends React.Component {
-  static propTypes = {
-    iconSrc: PropTypes.string.isRequired,
-    children: PropTypes.element.isRequired
-  }
+class HeaderDropDown extends React.Component<PropsType, StateType> {
+  toggleDropDown: Function
```

Comment:
What type of function?

Reply:
All of these are marked as warnings in flow. Can be fixed later. There are a lot of similar problems